src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-report.component.ts

```diff
@@ -90,5 +94,22 @@ export class GitDiffReportComponent implements OnInit {
         });
         this.leftCommit = this.report.leftCommitHash?.substring(0, 10);
         this.rightCommit = this.report.rightCommitHash?.substring(0, 10);
+        this.filePaths.forEach((path) => {
+            if (this.entriesByPath.get(path)?.length) {
+                this.diffsReadyByPath[path] = false;
+            }
+        });
+        this.nothingToDisplay = Object.keys(this.diffsReadyByPath).length === 0;
+    }
+
+    /**
+     * Records that the diff editor for a file has changed its "ready" state.
+     * If all paths have reported that they are ready, {@link allDiffsReady} will be set to true.
+     * @param path The path of the file whose diff this event refers to.
+     * @param ready Whether the diff is ready to be displayed or not.
+     */
+    onDiffReady(path: string, ready: boolean) {
+        this.diffsReadyByPath[path] = ready;
+        this.allDiffsReady = Object.values(this.diffsReadyByPath).reduce((a, b) => a && b, true);
```

Comment:
Is the default value necessary here? I think is is redundant in that case. IMO, it makes the code less clean and concise.

```suggestion
        this.allDiffsReady = Object.values(this.diffsReadyByPath).reduce((a, b) => a && b);
```

Reply:
I added the initial value because it makes the code more robust against errors (`reduce` without an initial value throws an error on an empty array). Since this won't be the case when we call this method, I'll remove the default value